### PR TITLE
Sriharsha/reduction deployment patch

### PR DESCRIPTION
**Pull request by sriharsha219**
Following changes are made as a patch update to execute the workflows and the respective shell scripts in the EC2 instance as 'ec2-user'
*Changes/additions are made to support the leto reduction - cv2_jpg_reduction, ffmpeg_resolution_downsampler, and fps_bitrate modules deployment in target AWS EC2 instances using a common standard shell script which will be deployed in EC2 and triggered with a GitHub workflow.
*This also contains a new workflow to run the leto - python scripts, which are already deployed in the EC2 instance.

Changes are done in the following files:
----------------------------------------
1.In '.github/workflows/create-ec2.yml', updated the 'uses: aws-actions/configure-aws-credentials@v1' to 'uses: aws-actions/configure-aws-credentials@v2'
2.In '.github/workflows/deploy-reduction-modules-in-ec2.yml', updated 'ssm-user/ text to 'ec2-user'
3.In '.github/workflows/run-python-scripts-in-ec2.yml', updated the steps name and ssm commands
4.In 'deployment_scripts/README.md', added new 'Prerequisite' steps and also updated instruction numbers '1' & '2'
5.In 'deployment_scripts/deploy-reduction-modules-in-ec2.sh', updated 'ssm-user/ text to 'ec2-user' and removed '#Add libraries path to the environment variable PATH temporarily', '#Add libraries path to the environment variable 'PATH' permanently' sections

New files:
----------
1.Added new shell script 'deployment_scripts/run-python-scripts-in-ec2.sh' to support python script execution in EC2 instance as 'ec2-user'

### CODE REVIEW

File: .github/workflows/create-ec2.yml
Comment (vinayakSharm) [inline .github/workflows/create-ec2.yml:1]: good

File: .github/workflows/deploy-reduction-modules-in-ec2.yml [hunk 1]
```diff
@@ -44,13 +44,13 @@ jobs:
...
        aws-region: ${{ env.AWS_REGION }}
-    #Download deployment shell script from github to '/home/ssm-user' path in target AWS EC2 instance
+    #Download deployment shell script from github to '/home/ec2-user' path in target AWS EC2 instance
     - name: Download deployment script from github to EC2 instance
       run: |
         sudo apt-get -y install expect-dev
         unbuffer aws ssm start-session \
         --document-name 'AWS-StartNonInteractiveCommand' \
-        --parameters '{"command": ["curl -o /home/ssm-user/deploy-reduction-modules-in-ec2.sh https://raw.githubusercontent.com/DISHDevEx/leto/main/deployment_scripts/deploy-reduction-modules-in-ec2.sh"]}' \
+        --parameters '{"command": ["curl -o /home/ec2-user/deploy-reduction-modules-in-ec2.sh https://raw.githubusercontent.com/DISHDevEx/leto/sriharsha/reduction-deployment-patch/deployment_scripts/deploy-reduction-modules-in-ec2.sh"]}' \
```
Comment (vinayakSharm) [inline .github/workflows/deploy-reduction-modules-in-ec2.yml]: do you want to curl from main?

Comment (vinayakSharm) [reply .github/workflows/deploy-reduction-modules-in-ec2.yml]: this branch will be deleted post pr

File: .github/workflows/run-python-scripts-in-ec2.yml [hunk 2]
```diff
@@ -39,10 +39,17 @@ jobs:
...
        aws-region: ${{ env.AWS_REGION }}
     #Run python file in target AWS EC2 instance
-    - name: Run python file in EC2 instance
+    - name: Download shell script from github to EC2 instance
       run: |
         sudo apt-get -y install expect-dev
         unbuffer aws ssm start-session \
         --document-name 'AWS-StartNonInteractiveCommand' \
-        --parameters '{"command": ["python3 /home/ssm-user/leto/${{ github.event.inputs.python_file_path }}"]}' \
+        --parameters '{"command": ["curl -o /home/ec2-user/run-python-scripts-in-ec2.sh https://raw.githubusercontent.com/DISHDevEx/leto/sriharsha/reduction-deployment-patch/deployment_scripts/run-python-scripts-in-ec2.sh"]}' \
```
Comment (vinayakSharm) [inline .github/workflows/run-python-scripts-in-ec2.yml]: do you want to curl from main?

File: .github/workflows/deploy-reduction-modules-in-ec2.yml (hunk 1 above)
Comment (sriharsha219) [reply .github/workflows/deploy-reduction-modules-in-ec2.yml]: Yes I just updated this link, pls refresh and check.

**Review by vinayakSharm — CHANGES_REQUESTED** [review]
change in the inputs to curl from main, other than that looks good.

**Comment by sriharsha219** [comment]
> change in the inputs to curl from main, other than that looks good.

Yes, I just updated the curl links in both yaml files. Pls reload and check.

**Review by vinayakSharm — APPROVED** [review]

**Pull request merged**